components/table/src/simple-columns/favorite-column.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright 2019 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Optional,
+  Output,
+  ViewEncapsulation,
+} from '@angular/core';
+import { DtTable } from '../table';
+import { DtSimpleColumnBase } from './simple-column-base';
+
+@Component({
+  selector: 'dt-favorite-column',
+  templateUrl: 'favorite-column.html',
+  styleUrls: ['./favorite-column.scss'],
+  preserveWhitespaces: false,
+  encapsulation: ViewEncapsulation.Emulated,
+  /*
+   * Deliberatley set to Default because the embedded view that gets created for the
+   * dtColumDef can't handle onPush and results in ChangeAfterChecked error.
+   */
+  changeDetection: ChangeDetectionStrategy.Default,
+  providers: [{ provide: DtSimpleColumnBase, useExisting: DtFavoriteColumn }],
+})
+export class DtFavoriteColumn<T> extends DtSimpleColumnBase<T> {
+  @Output() readonly change = new EventEmitter<T>();
```

Comment:
* We try to avoid (generic) outputs that are like native events (such like `change`) on components (except the ones that replace native elements, like checkbox).
* As it is public api, please provide a doc comment.

Reply:
Event was changed to `favoriteToggled` and the doc comment was added